daemon/logs_clear.go

```diff
@@ -0,0 +1,28 @@
+package daemon // import "github.com/docker/docker/daemon"
+
+import (
+	"os"
+
+	"github.com/docker/docker/errdefs"
+)
+
+func (daemon *Daemon) ContainerLogsClear(containerName string) error {
+	container, err := daemon.GetContainer(containerName)
+	if err != nil {
+		return errdefs.System(err)
+	}
+
+	f, err := os.OpenFile(container.LogPath, os.O_WRONLY, 0755)
```

Comment:
Couple of issues here:

1. "LogPath" is only used by the json log driver ("local", "journald" could use this, but wouldn't work like this)
2. "LogPath" is only one file but there can be multiple files (rotated files)
3. This will almost certainly break any logs being written and cause issues trying to read later.

This functionality really needs to live in the log driver so it can synchronize writes as well as handle other differences in implementations between them.
I also like how [journald handles this](https://www.freedesktop.org/software/systemd/man/latest/journalctl.html#--vacuum-size=).
It would be nice if we can expose similar functionality, or at least leave it open for similar functionality.